File: python/language/python-pratice/innerclass.py
class Human:
    def __init__(self):
        self.name = 'Guido'
        self.head = self.Head()
        self.brain = self.Brain()

    class Head:
        def talk(self):
            return 'talking...'

    class Brain:
        def think(self):
            return 'thinking...'

class Outer:
    num = 30
    def __init__(self):
        print ('Instance of outer class is created')

    class Inner:
        def __init__(self):
            print ('Instance of Inner class is created')

if __name__ == '__main__':
    guido = Human()
    print(guido.name)
    print(guido.head.talk())
    print(guido.brain.think())

    o = Outer.Inner()

    head = Human.Head()
    brian = Human.Brain()

File: java/language/src/inheritance/_InnerClass.java
package language.inheritance;

class Outer {
    public class Inner {
        void print(){
            System.out.println("Inner");
        }
        //static  void p() {
        //    System.out.println("Inner");
        //}
    }

    public static class sInner{
        static void print(){
            System.out.println("sInner static print");
        }
        void p(){
            System.out.println("sInner p");
        }
    }

    public static void main(String args[]) {
       Outer o = new Outer();
       Outer.sInner.print();
       Inner i = (new Outer()).new Inner();
       i.print();
    }

}